src/main/java/core/basesyntax/WorkWithFile.java

```
@@ -1,7 +1,58 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    static final String SUPPLY_NAME = "supply";
+    static final String BUY_NAME = "buy";
+    static final String COMMA = ",";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        int supplyAmount = 0;
+        int buyAmount = 0;
+        String operation;
+        int amountStr;
+        String[] linePart;
+
+        try (BufferedReader reader = new BufferedReader(new FileReader(fromFileName))) {
+            String value = reader.readLine();
+
+            while (value != null) {
+                linePart = value.split(COMMA);
+                if (linePart.length == 2) {
+                    operation = linePart[0].trim();
+                    amountStr = Integer.parseInt(linePart[1].trim());
```

Parsing the amount uses `Integer.parseInt(...)` but there is no handling for a malformed number. The checklist warns: "Don't ignore exceptions ... Better re-throw `RuntimeException` with the original exception and some message in the parameters: catch (Exception e) { throw new RuntimeException("Can't read data from the file " + fileName, e); }". Catch `NumberFormatException` (or validate before parsing) and rethrow a RuntimeException with a message that includes `fromFileName` and the original cause.